pkg/server/metrics.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2025 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	registry = prometheus.NewRegistry()
+
+	logIndexVerificationTotal = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "log_index_verification_total",
+		Help: "Total number of log consistency check attempts.",
+	})
+	logIndexVerificationFailure = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "log_index_verification_failure",
+		Help: "Total number of failed log consistency check attempts.",
+	})
+
+	signalChan = make(chan os.Signal, 1)
+)
+
+func init() {
+	registry.MustRegister(logIndexVerificationTotal, logIndexVerificationFailure)
+	signal.Notify(signalChan, os.Interrupt, syscall.SIGTERM)
+}
+
+// InitRegistryForTesting resets the registry for test isolation.
+// This is only used in tests to ensure a clean state.
+func InitRegistryForTesting() {
+	registry = prometheus.NewRegistry()
+	registry.MustRegister(logIndexVerificationTotal, logIndexVerificationFailure)
+}
+
```

Comment:
would it be better to put this elsewhere to separate regular code from testing code?